nco/custom.py

```diff
@@ -0,0 +1,268 @@
+            it=iter(Value)    
+            inputType=type(next(it))   
+            bIterable=True
+            del it   
+        except Exception:
+            bIterable=False 
+            inputType=type(Value) 
+         
+         
+        if sType:
+            try:    
+                npType=npTypesNP[sType] 
+            except:
+                raise Exception('{0}: specified Type "{1}" not found.\nValid values {2}\n'.format("atted()", sType, npTypes.keys()))                          
+        # set default types
+        else: 
+           if inputType is int:
+               npType=np.int32
+           elif inputType is float:
+               npType=np.float64
+           elif inputType is str:
+               npType=str
+           # check the input type 
+           else:  
+               try: 
+                   x=npTypes[str(np.dtype(inputType)) ] 
+                   npType=inputType
+               except:
+                   raise Exception('{0}: The type of Value "{1}" is NOT valid\nValid values {2}\n'.format("atted()", inputType, npTypes.keys()))                          
+
+        if bIterable:              
+            if npType is str:                          
+                # convert everything to string
+                sList=[ np.dtype(str).type(v) for v in Value]
+               
+                # get max string length
+                lenMax=len ( max(sList,key=len) )
+                # create array of 'fixed' length strings
+                npValue=np.array( sList , 'S{0}'.format(lenMax) )          
+       
+                # for v in Value:    
+                #     npValue.append(np.dtype(npType).type(v))
+            else: 
+                npValue=np.fromiter(Value, npType)  # create array from iterable all in one go !!
+        else:
+            npValue=np.dtype(npType).type(Value) 
+
+        
+        if npValue is not None:
+          self.npType=npType
+          self.npValue=npValue 
+                        
+    def __str__(self):
+          return ('mode="{0}" attName="{1}" varName="{2} Value="{3}" type="{4}"\n'.format(self.mode, self.attName, self.varName, self.npValue, self.npType))
+
+    def prnOption(self): 
+
+              
+        # modeChar=validModes[self.mode]
+        # deal with delete - nb doesnt need any data 
+        if self.mode=='d':  
+            return ('-a "{0}","{1}",{2},,'.format(self.attName,self.varName, self.mode))   
+
+        bList= type(self.npValue) is list or type(self.npValue) is np.ndarray       
+  
+        # deal with string quirks here 
+        if self.npType is str:
+           if bList:
+             typeChar="sng"
+           else:
+             typeChar="c"
+        else:
+           npTypeStr=str(np.dtype(self.npType)) 
+           typeChar=npTypes[npTypeStr]    
+
+
+        # deal with a single value first   
+        if not bList:
+           self.npValue=[self.npValue]
+
+        if self.npType is str:          
```

Comment:
It just occurred to me that you may need to be a bit more careful about your comparisons to the `str` datatype. There are some notable differences between Python 2 and 3 in this regard.

```Python
isinstance(self.np_type, (str, basestring))
```